neffint/fixed_grid_fourier_integral.py

```diff
@@ -0,0 +1,368 @@
+            phi_remaining_terms_bound = 2 * abs_x_to_n_plus_1 * exp_abs_x * inv_factorial / ((n+1)*(n+5))
+            phi_converged = np.all(phi_remaining_terms_bound < rel_tolerance * min_phi_component)
+        
+        # Only calculate pis if not yet converged
+        if not psi_converged:
+            psi[taylor_mask] += - 1/((n+3)*(n+4)) * jx_to_n * inv_factorial
+
+            # Check for convergence
+            min_psi_component = np.min([np.abs(np.real(psi[taylor_mask])), np.abs(np.imag(psi[taylor_mask]))], axis=0)
+            psi_remaining_terms_bound = abs_x_to_n_plus_1 * exp_abs_x * inv_factorial / ((n+1)*(n+4)*(n+5))
+            psi_converged = np.all(psi_remaining_terms_bound < rel_tolerance*min_psi_component)
+        
+        # Update variables
+        jx_to_n *= 1j*xx
+        abs_x_to_n_plus_1 *= np.abs(xx)
+        inv_factorial /= n+1
+
+    # If loop finished without converging, raise error
+    if not (phi_converged and psi_converged):
+        raise RuntimeError(f"Phi and Psi calculation failed to converge after {MAX_TAYLOR_ITERATIONS} iterations")
+    
+    return phi, psi
+
+def _fourier_integral_inf_correction(times: np.ndarray, omega_end: float, func_value_end: ArrayLike, func_derivative_end: ArrayLike=0.):
+    """Calculate the asymptotic correction term as omega->inf of a Fourier integral for the given times.
+    Uses a first or second (if func_derivative_end is given) order Taylor expansion around the angular frequency omega_end.
+
+    :param times: A 1D array of length M with the times to compute the Fourier integral for
+    :type times: np.ndarray
+    :param omega_end: A frequency to expand the Taylor series from
+    :type omega_end: float
+    :param func_value_end: An array of shape (X1, X2, ...) containing the values of the function to be transformed evaluated at omega_end
+    :type func_value_end: ArrayLike
+    :param func_derivative_end: An array of shape (X1, X2, ...) containing the derivative of the function to be transformed, evaluated at omega_end, defaults to 0.
+    :type func_derivative_end: ArrayLike, optional
+    :return: The asymptotic terms of the Fourier Integral for all times, given as an array of shape (M, X1, X2, ...)
+    :rtype: np.ndarray
+    """
+
+    # TODO: Reconsider func_value_end and func_derivative_end types, shapes and type hints
+    
+    func_value_end = np.asarray(func_value_end)
+    func_derivative_end = np.asarray(func_derivative_end)
+    
+    # Add axes from function output
+    for _ in range(func_value_end.ndim):
+        times = times[..., np.newaxis]
+    
+    # Add time axis
+    func_value_end = func_value_end[np.newaxis, ...]
+    func_derivative_end = func_derivative_end[np.newaxis, ...]
+    
+    return np.exp(1j*times*omega_end) * (1j*func_value_end/times - func_derivative_end/times**2)
+
+def fourier_integral_fixed_sampling(
+    times: ArrayLike,
+    frequencies: ArrayLike,
+    func_values: ArrayLike,
+    inf_correction_term: bool,
```

Comment:
Now that I think about it, it feels strange to have `inf_correction_term` as an option in the public interface. Presumably you would always want to use it.

The obvious exception is the adaptive algorithm, but that could use a hidden `_fourier_integral_fixed_sampling` function, and then the public version of the function wraps the private one but with `inf_correction_term` (and "`neg_inf_correction_term`", if we add that) set to `True` when they should be. I.e. if `frequencies[0] < 0`, `neg_inf_correction_term = True`, and if `frequencies[-1] > 0`, `inf_correction_term = True`.

Reply:
not necessarily.

I think it's fine as it is. Maybe one wants to compute a Fourier integral on a restricted range, after all, why not (I think it may even correspond to the way some signal are obtained - infinite bandwidth (or infinite times) do not really exist in reality...